### PR TITLE
Added support for  auav X2.1

### PR DESCRIPTION
**Pull request by davids5**
AKA as auav-x21

I had a hell of a time with the name auav-x2.1 it just killed the build in the .msg compilation 
```<command-line>:0:26: error: ISO C++11 requires whitespace after the macro name [-Werror]``` 
and px4fmu-v2_1  was truncated to px4fmu-v2 which led to multiply defined symbols. 

So the name is auav-x21 

This rev of the board may not prove to be FMUv2 compatible - it has no external SPI bus or the external I2C (I2C2) and I2C1 is used as the external I2C.

### CODE REVIEW

**Comment by kd0aij** [comment]
@davids5 Is there a binary for the X2.1 bootloader on github, or should I build it myself?

**Comment by davids5** [comment]
@kd0aij Please build it.

**Comment by kd0aij** [comment]
I get no response from the new bootloader when trying to flash (either the binary you sent or the one I built from PX4/Bootloader/master):
```
make auav-x21_default upload
PX4 CONFIG: /home/markw/git/kd0aij/PX4/Firmware/build_auav-x21_default
ninja: Entering directory `/home/markw/git/kd0aij/PX4/Firmware/build_auav-x21_default'
[1/1] uploading /home/markw/git/kd0aij/PX4/Firmware/build_auav-x21_default/src/firmware/nuttx/nuttx-auav-x21-default.px4
Loaded firmware for 21,0, size: 1080184 bytes, waiting for the bootloader...
If the board does not respond within 1-2 seconds, unplug and re-plug the USB connector.
attempting reboot on /dev/serial/by-id/pci-3D_Robotics_PX4_FMU_v2.x_0-if00...
if the board does not respond, unplug and re-plug the USB connector.
```
My procedure for flashing the bootloader seems to be working though; if I reflash the v2 bootloader:
```
make auav-x21_default upload
PX4 CONFIG: /home/markw/git/kd0aij/PX4/Firmware/build_auav-x21_default
ninja: Entering directory `/home/markw/git/kd0aij/PX4/Firmware/build_auav-x21_default'
[1/1] uploading /home/markw/git/kd0aij/PX4/Firmware/build_auav-x21_default/src/firmware/nuttx/nuttx-auav-x21-default.px4
Loaded firmware for 21,0, size: 1080184 bytes, waiting for the bootloader...
If the board does not respond within 1-2 seconds, unplug and re-plug the USB connector.
attempting reboot on /dev/serial/by-id/pci-3D_Robotics_PX4_FMU_v2.x_0-if00...
if the board does not respond, unplug and re-plug the USB connector.
Found board 9,0 bootloader rev 5 on /dev/serial/by-id/pci-3D_Robotics_PX4_BL_FMU_v2.x_0-if00
WARNING: Firmware not suitable for this board (board_type=9 board_id=33)
```

**Comment by davids5** [comment]
@kd0aij

 The boot load device instance is usb-AUAV_PX4_BL_AUAV_X2.1_0-if00 and  usb-AUAV_PX4_AUAV_X2.1_0-if00

I have not changed the upload script yet as indicated by:
 
https://github.com/PX4/Bootloader/blob/master/hw_config.h#L11-L17

I will push that change now - but I can not test it fully until next week and there may be other issues that need to be looked at.

File: src/drivers/px4io/px4io.cpp [hunk 1]
```diff
@@ -3625,7 +3625,8 @@ px4io_main(int argc, char *argv[])
 			fn[1] =	"/fs/microsd/px4io1.bin";
 			fn[2] =	"/fs/microsd/px4io.bin";
 			fn[3] =	nullptr;
-#elif defined(CONFIG_ARCH_BOARD_PX4FMU_V2)
```
Comment (LorenzMeier) [inline src/drivers/px4io/px4io.cpp]: @davids5 Can we move this into board config for all boards?

File: src/drivers/drv_gpio.h [hunk 2]
```diff
@@ -42,6 +42,34 @@
 
 #include <sys/ioctl.h>
 
+#ifdef CONFIG_ARCH_BOARD_AUAV_X21
```
Comment (LorenzMeier) [inline src/drivers/drv_gpio.h]: Can we move this into the board config?

**Comment by LorenzMeier** [comment]
@kd0aij Please work with Daniel and Don to get this board into all the build systems and QGC. In general its not a great idea to spin your own design as manufacturer for that very reason - it takes far too much effort to just add support for a single unit.

File: src/drivers/px4io/px4io.cpp (hunk 1 above)
Comment (davids5) [inline src/drivers/px4io/px4io.cpp]: Yes. I had a wip to do so but it got interrupted - this week is bad for me - but I will resume that work when I can.

**Comment by kd0aij** [comment]
@davids5 @LorenzMeier The AUAVX2.1 uses the same chip selects and I/O pins for int/ext SPI ports and has a single I2C port, exactly the same as the X2 and differing from fmu-v2 only in the lack of an "external" I2C port. 

Is a new bootloader necessary to support 2MB of flash, or is there info already present in the hardware version to determine the silicon rev?

**Comment by davids5** [comment]
Mark,

Unless I am missing something this is what I have, it not compatible for the noted reasons. 

image

I will push the fixes for the ICM is a second.

**Comment by kd0aij** [comment]
@LorenzMeier As far as I can tell, the none of the *_DRDY interrupt signals are actually used. Is that correct?

**Pull request merged**